code-dot-org/code-dot-org
# Measure contact rollups job

[PLC-258](https://codedotorg.atlassian.net/browse/PLC-258)

### What
- Time major steps in Contact Rollups job. Propagate results directly to `cron-daily` Slack channel ([example](https://codedotorg.slack.com/archives/C3S8AJY9K/p1560413138006100)).
- Raise HoneyBadger error when validation fails so the job cannot fail silently.

### Why
- Contact Rollups is a long running and resource intensive job (~3h, cannot run directly on production db) and has no test. It sometime failed for more than 1 week without notice. It has been a major tech debt and we will soon have to re-architect it. 
- The first few steps are 
  - Time important steps in the process to identify bottlenecks.
  - Collect important metrics such as # of new/updated rows to identify which steps are still valuable to keep.

### How tested
- Test LogCollector `bundle exec ruby lib/test/cdo/test_log_collector.rb`
- Test run in `test` server
  - Check out (cherry-pick) files in feature branch into test machine
  - Comment out Syncing to Pardot section so changes happen only in local db.
  - Start a `screen` section so job will still run even connection to test server breaks.
  - Run `bin/cron/build_contact_rollups`. 
  - (Detach = `ctrl+a+d`, resume section = `screen -r <sessions_name>`, list = `screen -ls`)

### Note
I add all PLC to this PR since Contact Rollups is in PLC land now and anyone of us may have to work on it in the future.

### TO DO
- [x] Remove stdout prints before merging.
